projectile.el

```diff
@@ -3091,10 +3091,17 @@ it acts on the current project."
 
 (defun projectile--cmake-command-presets (filename command-type)
   "Get CMake COMMAND-TYPE presets from FILENAME."
-  (when-let ((preset (projectile--cmake-read-preset (projectile-expand-root filename))))
-    (cl-remove-if
-     (lambda (preset) (equal (gethash "hidden" preset) t))
-     (gethash (projectile--cmake-command-preset-array-id command-type) preset))))
+  (when-let ((filename (projectile-expand-root filename))
+             (preset (projectile--cmake-read-preset filename)))
+    (append
+     (cl-remove-if
+      (lambda (preset) (equal (gethash "hidden" preset) t))
+      (gethash (projectile--cmake-command-preset-array-id command-type) preset))
+     (mapcar
```

Comment:
Perhaps you should make this is a separate private function, so it's clearer what it does exactly.

Reply:
I kept the old implementation unchanged, just changed the name (added `-shallow` to name). Now `projectile--cmake-command-presets` calls the old one, and contains only new stuff. Should be more clear.